sstables/storage.cc

```diff
@@ -73,7 +120,9 @@ class filesystem_storage final : public sstables::storage {
     virtual future<file> open_component(const sstable& sst, component_type type, open_flags flags, file_open_options options, bool check_integrity) override;
     virtual future<data_sink> make_data_or_index_sink(sstable& sst, component_type type) override;
     virtual future<data_sink> make_component_sink(sstable& sst, component_type type, open_flags oflags, file_output_stream_options options) override;
-    virtual future<> destroy(const sstable& sst) override { return make_ready_future<>(); }
+    virtual future<> destroy(const sstable& sst) override {
+        co_await _dir.destroy(sst);
+    }
```

Comment:
I don't understand why filesystem_storage::destroy() needs to change, just because we're changing some details about sync.

Reply:
This goes back to @xemul's [comment](https://github.com/scylladb/scylladb/pull/19624#discussion_r1670390612). I may have misunderstood that comment of course. Either way, I'm dropping the `destroy` modification from v5. (Similarly pointed out by @bhalevy [here](https://github.com/scylladb/scylladb/pull/19624#discussion_r1679970710).)

Reply:
addressed in v5, documented in the v5 changelog